### ISSUE TITLE
AVT 1 - Icon: DAP violations - undefined role

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#2486]
## Environment
macOS Mojave version 10.14.2
Chrome version 72.0.3626.96 (Official Build) (64-bit)

## Detailed Description
Run DAP on the icon component
See screenshot below for violations:
<img width="1244" alt="Screen Shot 2019-04-17 at 3 08 02 PM" src="https://user-images.githubusercontent.com/21676914/56317812-4ae85300-6123-11e9-88b1-462da8d19246.png">

Note: Add a valid ARIA role or remove the role entirely if it is not needed.

**Comment by snidersd** [comment]
@dakahn Adding role="img" will remove the violation, but it may not be correct depending on how an icon is used. IMO it also needs to be documented so developers understand that it may need to be changed. In most cases, the "+" icon that is displayed in the example is used as a button to expand and collapse web content. In that case, it would have a role="button" and a state of aria-expanded="false" (when closed). However, if the icon is used inside a <button> element then the <button> element would have the role and state and the icon can be ignored since it adds no value.

**Comment by dakahn** [comment]
icon by default should be `role=image`
